### PR TITLE
Fix Playground rewrite rules

### PR DESCRIPTION
**Pull request by sejas**
- Related to: #491 
p1725358414451329/1725354412.932809-slack-C04GESRBWKW
- Related to: https://github.com/Automattic/studio/issues/468

## Proposed Changes

- Add wordPressRewriteRules from `@wp-playground/wordpress`

Kudos to the whole @Automattic/yolo team for finding and fixing the issue.

## Testing Instructions

- Run 'npm install && npm start' 
- Create a page and set the permalink in the settings, or import a site with that setting
- Observe the page renders correctly instead of a "Not found" error.

about-me-w

## Pre-merge Checklist

- [ ] Have you checked for TypeScript, React or other console errors?

### CODE REVIEW

File: src/hooks/use-i18n-data.tsx [hunk 1]
```diff
@@ -22,6 +22,7 @@ export const I18nDataProvider = ( { children }: { children: React.ReactNode } )
 
 	const initI18n = useCallback( async ( localeKey: SupportedLocale ) => {
 		const newI18n = createI18n( getLocaleData( localeKey )?.messages );
+		defaultI18n.setLocaleData( getLocaleData( localeKey )?.messages );
```
Comment (fluiddot) [inline src/hooks/use-i18n-data.tsx]: This will be incorporated in a separate PR https://github.com/Automattic/studio/pull/504, so we could remove it.

File: vendor/wp-now/src/wp-now.ts [hunk 2]
```diff
@@ -61,7 +62,8 @@ export default async function startWPNow(
 		},
 		documentRoot: documentRoot || '/wordpress',
 		absoluteUrl: options.absoluteUrl,
-		rewriteRules: []
+		rewriteRules:wordPressRewriteRules,
```
Comment (fluiddot) [inline vendor/wp-now/src/wp-now.ts]: Nitpick: 
```suggestion
		rewriteRules: wordPressRewriteRules,
```

File: src/hooks/use-i18n-data.tsx (hunk 1 above)
Comment (sejas) [reply src/hooks/use-i18n-data.tsx]: My bad!, I've removed it! https://github.com/Automattic/studio/pull/505/commits/fdd67795a72c110814de4a5afa80b9ce370338a6

File: vendor/wp-now/src/wp-now.ts (hunk 2 above)
Comment (sejas) [reply vendor/wp-now/src/wp-now.ts]: Updated! https://github.com/Automattic/studio/pull/505/commits/f261bd3edd88b4d8013988e239bc5a7eec681dcb

**Review by fluiddot — APPROVED** [review]
LGTM 🎊 ! I've confirmed that using non-plain permalink structures works as expected. Great work team @Automattic/yolo !
